### PR TITLE
APP-13623 pdf reader thumbnails

### PR DESCRIPTION
**Pull request by Sqrrl**
## Summary

This also adds support for the PDF side-by-side view.

- [Ticket](https://flipapp.myjetbrains.com/youtrack/issue/APP-13623/Web-Component-Add-Page-Thumbnail-view-to-PDF-Reader)
- [Design](https://www.figma.com/file/iRHiAFsHxnzux8gQ9EjlZe/Swirl-Components?node-id=3114%3A19586&t=1QMb89z7VFXRSKal-0)
- [Storybook](https://gentle-water-054ae5203-166.westeurope.1.azurestaticapps.net/?path=/docs/components-swirlpdfreader--swirl-pdf-reader)

### General

- [ ] [Changeset added](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md)
- [ ] The submitted code is organized and formatted according to our [💅 Style Guide](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-style-guide--page).
  - [ ] The code is formatted with Prettier.
  - [ ] There are no linting errors.

### For new or updated components

https://swirl-storybook.flip-app.dev/?path=/docs/contributions-merge-publish--page

- [ ] The changes do not contain any breaking changes.
- [ ] The changes do not introduce new components that don't belong in the library (e.g. non-reusable components, highly specialized components, components including business logic)
- [ ] The component documentation is updated.
- [ ] The changes meet the [🤖 testing requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-testing--page).
  - [ ] New features are tested.
  - [ ] In case of bug fixes, regression tests have been added.
  - [ ] All tests are 🟢.
- [ ] The changes meet the [🧏‍♀️ accessibility requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-accessibility--page).
  - [ ] WCAG 2.1 Level A and Level AA requirements are met.
  - [ ] The Storybook a11y addon shows no errors.
  - [ ] The changes have been tested with a screen reader.
  - [ ] Keyboard controls have been tested, if applicable.
  - [ ] Components implementing form controls (inputs, buttons, selects, etc.) do not use Shadow DOM, and instead use Stencil's scoping mechanism
- [ ] The changes use our [🌈 theming concept](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-theming--page).
  - [ ] Design tokens have been used where appropriate.
  - [ ] The component has been visually checked in combination with the "Light" and "Dark" theme.
- [ ] The changes meet our [🌍 internationalization requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-internationalization--page).
  - [ ] No static text is used.
  - [ ] The component doesn't break with longer texts or different text wrappings.
  - [ ] Number, currency and date values can be formatted appropriately.
- [ ] The changes work in all supported browsers and viewports. See [📱 Responsive Design](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-responsive-design--page)

### CODE REVIEW

File: packages/swirl-components/src/components/swirl-pdf-reader/swirl-pdf-reader.css [hunk 1]
```diff
@@ -154,10 +166,48 @@
...
+  background-color: var(--s-background-default);
+  flex-direction: column;
+  transition: max-width 0.2s linear;
+  box-shadow: inset calc(-1 * var(--s-border-width-default)) 0 0 0
+    var(--s-border-default);
+  gap: var(--s-space-4);
+}
+
+.pdf-reader__thumbnail {
+  width: 4rem;
```
Comment (webspaceadam) [inline packages/swirl-components/src/components/swirl-pdf-reader/swirl-pdf-reader.css:195]: (question): is 4 correct here? in the design i see 5rem / 80px

File: packages/swirl-components/src/components/swirl-pdf-reader/swirl-pdf-reader.css [hunk 2]
```diff
@@ -1,6 +1,8 @@
 @import "../../styles/media-queries.css";
 
 :host {
+  --thumbnails-width: 6rem;
```
Comment (webspaceadam) [inline packages/swirl-components/src/components/swirl-pdf-reader/swirl-pdf-reader.css:4]: (issue): in the design the thumbnail container is a little bit bigger. with 112 px which should be 7.

**Review by webspaceadam — CHANGES_REQUESTED** [review]
Seems that the padding is missing?

image

vs


image

**Review by webspaceadam — COMMENTED** [review]
Its just unclear with the paddings and stuff of the thumbnails

File: packages/swirl-components/src/components/swirl-pdf-reader/swirl-pdf-reader.css (hunk 1 above)
Comment (Sqrrl) [reply packages/swirl-components/src/components/swirl-pdf-reader/swirl-pdf-reader.css]: Fixed.

File: packages/swirl-components/src/components/swirl-pdf-reader/swirl-pdf-reader.css (hunk 2 above)
Comment (Sqrrl) [reply packages/swirl-components/src/components/swirl-pdf-reader/swirl-pdf-reader.css]: Fixed.

**Comment by Sqrrl** [comment]
Cannot reproduce the padding issue. Any specific browser or viewport? Looks like this on my machine:
Bildschirm­foto 2023-02-17 um 15 07 11

**Pull request merged**